syntax = "proto3";

package proto;

service Api {
    rpc Send(SendRequest) returns (SendResponse) {}
    rpc Poll(PollRequest) returns (PollResponse) {}
}

message Header {
    bytes key = 1;
    bytes value = 2;
}

message Message {
    repeated Header headers = 1;
    optional bytes key = 2;
    bytes value = 3;
}

message SendRequest {
    string topic = 1;
    Message message = 2;
}

message SendResponse {
    uint64 partition = 1;
    uint64 offset = 2;
}

message PollRequest {
    repeated Header headers = 1;
    bytes groupId = 2;
    bytes clientId = 3;
    string topic = 4;
    uint64 maxCount = 5;
}

message PollResponse {
    repeated Message messages = 1;
}
